src/code/z_camera.c

```diff
@@ -8,12 +8,24 @@ s32 Camera_ChangeModeFlags(Camera* camera, s16 mode, u8 flags);
 s32 Camera_QRegInit(void);
 s32 Camera_UpdateWater(Camera* camera);
 
-#define RELOAD_PARAMS \
-    (camera->animState == 0 || camera->animState == 0xA || camera->animState == 0x14 || R_RELOAD_CAM_PARAMS)
+#define CAM_GET_RO_DATA(type) &((type*)camera->paramData)->roData // Read-Only Data
+#define CAM_GET_RW_DATA(type) &((type*)camera->paramData)->rwData // Read-Write Data
 
-#define PCT(x) ((x)*0.01f)
-#define NEXTSETTING ((values++)->val)
-#define NEXTPCT PCT(NEXTSETTING)
+// Camera will reload its paramData. This is mainly read-only data from camera_data. Although some read-write data is
+// reset as well
+#define RELOAD_PARAMS (camera->animState == 0 || camera->animState == 10 || camera->animState == 20)
+
+/**
+ * Camera data is stored in both read-only data and OREG as s16, and then converted to the appropriate type during
+ * runtime. If a small f32 is being stored as an s16, it is common to store that value 100 times larger than the
+ * original value. This is then scaled back down during runtime with the CAM_DATA_SCALED macro.
+ */
+#define CAM_DATA_SCALED(x) ((x)*0.01f)
+
+// Load the next setting from camera_data.c
+#define GET_NEXT_RO_DATA ((values++)->val)
+// Load the next setting from camera_data.c and scale
```

Comment:
Mentioning camera_data.c is a bit weird when you think about the rom post-linking :sweat_smile: 
It could also be a bit more specific I guess since camera_data is so all over the place with various types
I think in particular here it's reading from the mode-per-setting-specific `CameraModeValue` arrays?

Reply:
Correct, it's reading from `CameraModeValue`. I'll change the comment to focus on the data rather than the file